ibm/data_source_ibm_kms_key.go

```diff
@@ -185,6 +185,10 @@ func dataSourceIBMKMSKeyRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	instanceID := d.Get("instance_id").(string)
+	CrnInstanceID := strings.Split(instanceID, ":")
+	if len(CrnInstanceID) > 3 {
+		instanceID = CrnInstanceID[len(CrnInstanceID)-3]
```

Comment:
hs-crypto references is not removed in all files..
Please make sure you set client url similar to kms_key resource